component/pipeline.go

```diff
@@ -0,0 +1,85 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package component // import "go.opentelemetry.io/collector/component"
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+type PipelineID struct {
+	typeVal DataType `mapstructure:"-"`
+	nameVal string   `mapstructure:"-"`
```

Comment:
This should be unnecessary, private fields are not modified by mapstructure (reflection in Go does not allow that):

```suggestion
	typeVal DataType
	nameVal string
```